src/VisualStudio/CSharp/Impl/Options/Formatting/FormattingOptionPageControl.xaml.cs

```diff
@@ -65,40 +65,17 @@ public FormattingOptionPageControl(IServiceProvider serviceProvider) : base(serv
             BindToOption(RemoveUnusedVariablesCheckBox, CodeCleanupOptions.RemoveUnusedVariables, LanguageNames.CSharp);
         }
 
-        private void FormatWhenTypingCheckBox_Checked(object sender, RoutedEventArgs e)
-        {
-            FormatOnCloseBraceCheckBox.IsChecked = true;
-            FormatOnSemicolonCheckBox.IsChecked = true;
-
-            SetNestedCheckboxesEnabled();
-        }
-
-        private void FormatWhenTypingCheckBox_Unchecked(object sender, RoutedEventArgs e)
-        {
-            FormatOnCloseBraceCheckBox.IsChecked = false;
-            FormatOnSemicolonCheckBox.IsChecked = false;
-
-            SetNestedCheckboxesEnabled();
-        }
-
-        private void SetNestedCheckboxesEnabled()
-        {
-            FormatOnCloseBraceCheckBox.IsEnabled = FormatWhenTypingCheckBox.IsChecked == true;
-            FormatOnSemicolonCheckBox.IsEnabled = FormatWhenTypingCheckBox.IsChecked == true;
-        }
-
-        internal void SetCodeCleanupAsConfigured()
+        internal override void SaveSettings()
         {
-            var areCodeCleanupRulesConfigured = OptionService.GetOption<bool>(CodeCleanupOptions.AreCodeCleanupRulesConfigured, LanguageNames.CSharp);
+            base.SaveSettings();
 
-            if (!areCodeCleanupRulesConfigured)
-            {
-                var oldOptions = OptionService.GetOptions();
-                var newOptions = oldOptions.WithChangedOption(CodeCleanupOptions.AreCodeCleanupRulesConfigured, LanguageNames.CSharp, true);
+            // once formatting option is set, we never show code cleanup info bar again
```

Comment:
:memo: This comment is not completely correct, since we're setting the "Never show" flag even if they did not enable code cleanup.

Reply:
by formatting option is set, I didn't mean codeclean option is on, I meant, if user clicked "Okay" button on formatting option page regardless whether user enabled/disabled code cleanup option.

this behavior is same as original PR. original PR just had 2 options (option configured and never show info bar) which basically do same thing. so I merged 2 to 1.